docs/scripting-element-object.rst

```diff
@@ -0,0 +1,744 @@
+**Async:** no.
+
+
+.. _splash-element-text:
+
+element:text
+------------
+
+Fetch a text information from the element
+
+**Signature:** ``text = element:text()``
+
+**Returns:** ``text`` is a text content
+of the element.
+
+**Async:** no.
+
+It tries to return the trimmed value of the following JavaScript ``Node`` properties:
+
+* textContent
+* innerText
+* value
+
+If all of them are empty an empty string is returned.
+
+
+.. _splash-element-info:
+
+element:info
+------------
+
+Get useful information about the element.
+
+**Signature:** ``info = element:info()``
+
+**Returns:** ``info`` is a table with element info.
+
+**Async:** no.
+
+Info is a table with the following fields:
+
+* nodeName - node name in a lower case (e.g. *h1*)
+* attributes - table with attributes names and its values
+* tag - html string representation of the element
+* html - inner html of the element
+* text - inner text of the element
+* x - x coordinate of the element
+* y - y coordinate of the element
+* width - width of the element
+* height - height of the element
+* visible - flag representing if the element is visible
+
+
+.. _splash-element-field-value:
+
+element:field_value
+-------------------
+
+Get value of the field element (input, select, textarea, button).
+
+**Signature:** ``ok, info = element:field_value()``
+
+**Returns:** ``ok, info`` pair. If ``ok`` is nil then error happened during the function call;
+``value`` provides an information about error type; otherwise ``info`` is a value of the
+element.
+
+**Async:** no.
+
+This methods in the following way:
+
+    - if the element type is ``select``:
+        - if the ``multiple`` attribute is ``true`` it returns a *table* with the selected values;
+        - otherwise it returns the value of the select;
+    - if the element has attribute ``type="radio"``:
+        - if it's checked returns its value;
+        - other it returns ``nil``
+    - if the element has attribute ``type="checkbox"`` it returns *bool* value
+    - otherwise it returns the value of the ``value`` attribute or *empty string* if it doesn't exist
+
+
+.. _splash-element-form-values:
+
+element:form_values
+-------------------
+
+Return a table with form values if the element type is *form*
+
+**Signature:** ``ok, info = element:form_values()``
+
+**Returns:** ``ok, values`` pair. If ``ok`` is nil then error happened during the function call
+or node type is not *form*; ``values`` provides an information about error type; otherwise
+``values`` is a table with element names as keys and values as values.
+
+**Async:** no.
+
+Example: return the values of the following login form
+
+.. code-block:: html
+
+    <form id="login">
+        <input type="text" name="username" value="admin" />
+        <input type="password" name="password" value="pass" />
+    </form>
+
+.. code-block:: lua
+
+    function main(splash)
+        -- ...
+        local form = splash:select('#login')
+        local ok, values = assert(form:form_values())
```

Comment:
What do you think about removing 'ok' flag, and returning nil, reason instead of values table if there is an error? This is the way [splash:wait_for_resume](http://splash.readthedocs.io/en/stable/scripting-ref.html#splash-wait-for-resume) works; it allows to write just `value = assert(form:form_values())`